### PR TITLE
Move content of storage/tsdb to storage/backend(cortexproject#2262)

### PR DESCRIPTION
**Pull request by jaybatra26**
To make code support of backends(azure, s3 etc). The contents of
storage/tsdb was moved to storage/backend
This is an incremental PR as mentioned in #2262. 
Signed off by: Jay Batra <jaybatra73@gmail.com>

### CODE REVIEW

**Comment by jaybatra26** [comment]
@jtlisi and @pracucci PTAL.

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@
   * `-flusher.wal-dir` for the WAL directory to recover from.
   * `-flusher.concurrent-flushes` for number of concurrent flushes.
   * `-flusher.flush-op-timeout` is duration after which a flush should timeout.
+* [ENHANCEMENT] Move files from /storage/tsdb to a more generic folder i.e /storage/backend. #2262
```
Comment (bboreham) [inline CHANGELOG.md]: Changelog is for things that may impact an end-user; it's better to leave out internal refactoring to make the document smaller hence easier to read.

Comment (jaybatra26) [reply CHANGELOG.md]: @bboreham Will update this to something that makes better sense.

**Comment by jaybatra26** [comment]
@bboreham On consolidating, I think @pracucci's inputs would be highly appreciated here.

File: CHANGELOG.md (hunk 1 above)
Comment (jtlisi) [reply CHANGELOG.md]: @jaybatra26 No need to add anything to the changelog.

I also agree with @bboreham, we should move those packages as well. Up to you Jay if you want to do that in this PR.

@bboreham #2262 lays out the reasoning behind this. This PR is the first step in a series of PRs to reuse an external object client rolling our own.

**Comment by jtlisi** [comment]
@jaybatra26 If you need help setting up hooks for DCO signing, this page here has good info: https://github.com/prometheus/prometheus/wiki/DCO-signing

File: pkg/storage/backend/ref_cache.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package tsdb
+package backend
```
Comment (pracucci) [inline pkg/storage/backend/ref_cache.go]: This is `tsdb` specific. Please do not move it.

File: pkg/storage/backend/user_bucket_client.go [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package tsdb
+package backend
```
Comment (pracucci) [inline pkg/storage/backend/user_bucket_client.go]: This is `tsdb` specific. Please do not move it.

File: pkg/storage/backend/index_cache.go [hunk 4]
```diff
@@ -1,4 +1,4 @@
-package tsdb
+package backend
```
Comment (pracucci) [inline pkg/storage/backend/index_cache.go]: This is `tsdb` specific. Please do not move it.

File: pkg/storage/backend/config.go [hunk 5]
```diff
@@ -1,4 +1,4 @@
-package tsdb
+package backend
```
Comment (pracucci) [inline pkg/storage/backend/config.go]: This is `tsdb` specific. Please do not move it.

File: pkg/ingester/ingester_v2.go [hunk 6]
```diff
@@ -24,7 +24,7 @@ import (
 
 	"github.com/cortexproject/cortex/pkg/ingester/client"
 	"github.com/cortexproject/cortex/pkg/ring"
-	cortex_tsdb "github.com/cortexproject/cortex/pkg/storage/tsdb"
+	cortex_tsdb "github.com/cortexproject/cortex/pkg/storage/backend"
```
Comment (pracucci) [inline pkg/ingester/ingester_v2.go]: May you remove the package renaming to `tsdb` at all, please? I mean changing this line to:
`"github.com/cortexproject/cortex/pkg/storage/backend"`

And then fix the package references across the file.

_Same comment applies across the whole PR._

**Comment by jaybatra26** [comment]
@jtlisi @pracucci and @bboreham PTAL

File: CHANGELOG.md (hunk 1 above)
Comment (bboreham) [reply CHANGELOG.md]: Sorry, I'm not used to looking in the title of a PR to see what issue(s) it relates to; I expect that in the description.

Comment (jaybatra26) [reply CHANGELOG.md]: @bboreham, I have updated the description.

**Comment by jaybatra26** [comment]
Thanks

**Pull request merged**